File: SDKs/amazonaws.com/ec2/2016-11-15/go/pkg/models/shared/getreservedinstancesexchangequoteresult.go
package shared

// GetReservedInstancesExchangeQuoteResultReservedInstanceValueRollup
// The cost associated with the Reserved Instance.
type GetReservedInstancesExchangeQuoteResultReservedInstanceValueRollup struct {
	HourlyPrice           map[string]interface{}
	RemainingTotalValue   map[string]interface{}
	RemainingUpfrontValue map[string]interface{}
}

// GetReservedInstancesExchangeQuoteResultTargetConfigurationValueRollup
// The cost associated with the Reserved Instance.
type GetReservedInstancesExchangeQuoteResultTargetConfigurationValueRollup struct {
	HourlyPrice           map[string]interface{}
	RemainingTotalValue   map[string]interface{}
	RemainingUpfrontValue map[string]interface{}
}

// GetReservedInstancesExchangeQuoteResult
// Contains the output of GetReservedInstancesExchangeQuote.
type GetReservedInstancesExchangeQuoteResult struct {
	CurrencyCode                        map[string]interface{}
	IsValidExchange                     map[string]interface{}
	OutputReservedInstancesWillExpireAt map[string]interface{}
	PaymentDue                          map[string]interface{}
	ReservedInstanceValueRollup         *GetReservedInstancesExchangeQuoteResultReservedInstanceValueRollup
	ReservedInstanceValueSet            map[string]interface{}
	TargetConfigurationValueRollup      *GetReservedInstancesExchangeQuoteResultTargetConfigurationValueRollup
	TargetConfigurationValueSet         map[string]interface{}
	ValidationFailureReason             map[string]interface{}
}

File: SDKs/amazonaws.com/ec2/2016-11-15/java/lib/src/main/java/openapisdk/models/shared/GetReservedInstancesExchangeQuoteResult.java
package openapisdk.models.shared;



/**
 * GetReservedInstancesExchangeQuoteResult
 * Contains the output of GetReservedInstancesExchangeQuote.
**/
public class GetReservedInstancesExchangeQuoteResult {
    public java.util.Map<String, Object> currencyCode;
    public GetReservedInstancesExchangeQuoteResult withCurrencyCode(java.util.Map<String, Object> currencyCode) {
        this.currencyCode = currencyCode;
        return this;
    }
    public java.util.Map<String, Object> isValidExchange;
    public GetReservedInstancesExchangeQuoteResult withIsValidExchange(java.util.Map<String, Object> isValidExchange) {
        this.isValidExchange = isValidExchange;
        return this;
    }
    public java.util.Map<String, Object> outputReservedInstancesWillExpireAt;
    public GetReservedInstancesExchangeQuoteResult withOutputReservedInstancesWillExpireAt(java.util.Map<String, Object> outputReservedInstancesWillExpireAt) {
        this.outputReservedInstancesWillExpireAt = outputReservedInstancesWillExpireAt;
        return this;
    }
    public java.util.Map<String, Object> paymentDue;
    public GetReservedInstancesExchangeQuoteResult withPaymentDue(java.util.Map<String, Object> paymentDue) {
        this.paymentDue = paymentDue;
        return this;
    }
    public GetReservedInstancesExchangeQuoteResultReservedInstanceValueRollup reservedInstanceValueRollup;
    public GetReservedInstancesExchangeQuoteResult withReservedInstanceValueRollup(GetReservedInstancesExchangeQuoteResultReservedInstanceValueRollup reservedInstanceValueRollup) {
        this.reservedInstanceValueRollup = reservedInstanceValueRollup;
        return this;
    }
    public java.util.Map<String, Object> reservedInstanceValueSet;
    public GetReservedInstancesExchangeQuoteResult withReservedInstanceValueSet(java.util.Map<String, Object> reservedInstanceValueSet) {
        this.reservedInstanceValueSet = reservedInstanceValueSet;
        return this;
    }
    public GetReservedInstancesExchangeQuoteResultTargetConfigurationValueRollup targetConfigurationValueRollup;
    public GetReservedInstancesExchangeQuoteResult withTargetConfigurationValueRollup(GetReservedInstancesExchangeQuoteResultTargetConfigurationValueRollup targetConfigurationValueRollup) {
        this.targetConfigurationValueRollup = targetConfigurationValueRollup;
        return this;
    }
    public java.util.Map<String, Object> targetConfigurationValueSet;
    public GetReservedInstancesExchangeQuoteResult withTargetConfigurationValueSet(java.util.Map<String, Object> targetConfigurationValueSet) {
        this.targetConfigurationValueSet = targetConfigurationValueSet;
        return this;
    }
    public java.util.Map<String, Object> validationFailureReason;
    public GetReservedInstancesExchangeQuoteResult withValidationFailureReason(java.util.Map<String, Object> validationFailureReason) {
        this.validationFailureReason = validationFailureReason;
        return this;
    }
}